src/Microsoft.Health.Fhir.Liquid.Converter/Filters/SegmentFilters.cs

```diff
@@ -128,5 +128,37 @@ public static bool HasSegments(Hl7v2Data hl7v2Data, string segmentIdContent)
 
             return result;
         }
+
+        public static List<Hl7v2Data> SplitDataBySegments(Hl7v2Data hl7v2Data, string segmentIdSeparators)
+        {
+            var results = new List<Hl7v2Data>();
+            var result = new Hl7v2Data();
+            var segmentIds = segmentIdSeparators.Split(@"|", StringSplitOptions.RemoveEmptyEntries);
+
+            if (segmentIdSeparators == string.Empty || segmentIds.Intersect(hl7v2Data.Meta).ToList().Count == 0)
+            {
+                results.Add(hl7v2Data);
+                return results;
+            }
+
+            for (var i = 0; i < hl7v2Data.Meta.Count; ++i)
+            {
+                if (segmentIds.Contains(hl7v2Data.Meta[i]))
+                {
+                    results.Add(result);
+                    result = new Hl7v2Data();
+                }
+
+                result.Meta.Add(hl7v2Data.Meta[i]);
+                result.Data.Add(hl7v2Data.Data[i]);
```

Comment:
Will the segment separator be the first segment in data, or the last segment in data, or excluded in data?

Reply:
The first segment in result. It will be noted in the doc later.